File: ChessAPI/Domain/Entities/Tournament.cs
using Domain.Enums;

namespace Domain.Entities;

public class Tournament
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Location { get; set; } = string.Empty;
    public int MinPlayers { get; set; } = 2;
    public int MaxPlayers { get; set; } 
    public int MinElo { get; set; } = 1000;
    public int MaxElo { get; set; }
    public TournamentStatus Status { get; set; } = TournamentStatus.WaitingForPlayers;
    public int CurrentRound { get; set; } = 0;
    public bool WomenOnly { get; set; }
    public DateTime RegistrationDeadline { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.Now;
    public DateTime UpdatedAt { get; set; } = DateTime.Now;

    // Ajout des relations pour correspondre au diagramme
    public List<Category> Categories { get; set; } = new();
    public List<Player> RegisteredPlayers { get; set; } = new();
}

File: ChessClient/src/app/models/tournament.ts
import { Category } from './category';
import { Match } from './match';
import { Player } from './player';

export interface Tournament {
    id: number;
    name: string;
    location: string;
    minPlayers: number;
    maxPlayers: number;
    minElo: number;
    maxElo: number;
    status: TournamentStatus;
    currentRound: number;
    womenOnly: boolean;
    registrationDeadline: string;
    createdAt: string;
    updatedAt: string;
    categories: Category[];
    registeredPlayers: Player[];
    currentRoundMatches: Match[];
}

export enum TournamentStatus {
    WaitingForPlayers = 0,
    InProgress = 1,
    Canceled = 2,
    Closed = 3
}